src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/service/PrisonerContactRegistryService.kt

```diff
@@ -8,17 +8,23 @@ import org.springframework.web.reactive.function.client.WebClientResponseExcepti
 import uk.gov.justice.digital.hmpps.prisonercontactregistry.client.PrisonApiClient
 import uk.gov.justice.digital.hmpps.prisonercontactregistry.dto.AddressDto
 import uk.gov.justice.digital.hmpps.prisonercontactregistry.dto.ContactDto
+import uk.gov.justice.digital.hmpps.prisonercontactregistry.dto.RestrictionDto
+import java.time.LocalDate
 import java.util.function.Supplier
 
 @Service
 class PrisonerContactRegistryService(private val prisonApiClient: PrisonApiClient) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+    const val BANNED_RESTRICTION_TYPE = "BAN"
+  }
 
-  fun getContactList(prisonerId: String, contactType: String? = null, personId: Long? = null, withAddress: Boolean? = true): List<ContactDto> {
+  fun getContactList(prisonerId: String, contactType: String? = null, personId: Long? = null, withAddress: Boolean? = true, approvedVisitorsOnly: Boolean? = null): List<ContactDto> {
```

Comment:
Can we add a log line for this function, similar to the new getApprovedSocialContactList() function.

Also should the parameter "approvedVisitorsOnly" be null? Would it be possible to have it default to false to avoid null issues?

Reply:
Updated